test/tangents.jl

```diff
@@ -163,51 +163,46 @@
     end
 end
 
-# TODO: ideally we want to add the following test to the above testset (defined src/tangent.jl)
-# but we have to delay this until `randn_tangent` is implemented and working.
-@testset "zero_tangent" begin
+# TODO: add the following test to `tangent_test_cases`
+@testset "zero_tangent and randn_tangent" begin
     @testset "circular reference" begin
-        foo = Tapir.TestResources.TypeUnstableMutableStruct(5.0, nothing)
-        foo.b = foo
+        foo = make_circular_reference_struct()
         zt = Tapir.zero_tangent(foo)
-        @test zt.fields.b === zt
+        @test zt.fields.b.tangent === zt
+        rt = Tapir.randn_tangent(Xoshiro(123456), foo)
+        @test rt.fields.b.tangent === rt
     end
 
     @testset "struct with non-concrete fields" begin
         bar = Tapir.TestResources.TypeUnstableStruct(5.0, 1.0)
-        @test Tapir.zero_tangent(bar) == Tangent{@NamedTuple{a::Float64, b}}(@NamedTuple{a::Float64, b}((0.0, 0.0)))
+        @test Tapir.zero_tangent(bar) == Tangent{@NamedTuple{a::PossiblyUninitTangent{Float64}, b::PossiblyUninitTangent{Any}}}((a=PossiblyUninitTangent{Float64}(0.0), b=PossiblyUninitTangent{Any}(0.0)))
```

Comment:
Formatting, maybe consider something like
```suggestion
@test ==(
    Tapir.zero_tangent(bar),
    Tangent{@NamedTuple{a::PossiblyUninitTangent{Float64}, b::PossiblyUninitTangent{Any}}}(
        (a=PossiblyUninitTangent{Float64}(0.0), b=PossiblyUninitTangent{Any}(0.0))
    )
)
```
I really need to set up formatting CI...